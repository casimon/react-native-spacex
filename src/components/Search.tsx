import React from 'react';
import {
  TextInput,
  View,
  StyleSheet,
  StyleProp,
  TextStyle,
  ViewStyle
} from 'react-native';

import colors from '../constants/colors';

const styles = StyleSheet.create({
  search: {
    padding: 16,
    paddingTop: 22
  },
  input: {
    height: 40,
    borderRadius: 20,
    paddingHorizontal: 30,
    backgroundColor: '#ccc'
  }
});

interface Props {
  text: string;
  style?: StyleProp<ViewStyle>;
  onChangeText: (text: string) => void;
}

class Search extends React.PureComponent<Props> {
  render() {
    return (
      <View style={[styles.search, this.props.style]}>
        <TextInput
          style={styles.input}
          onChangeText={this.props.onChangeText}
          value={this.props.text}
          allowFontScaling={false}
          autoCompleteType="off"
          autoCorrect={false}
          placeholder="Search launches e.g CRS-17"
          placeholderTextColor="#666"
        />
      </View>
    );
  }
}

export default Search;
