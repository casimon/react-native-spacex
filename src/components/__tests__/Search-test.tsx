/**
 * @format
 */

import 'react-native';
import React from 'react';
import Search from '../Search';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders search with empty search string', () => {
  const tree = renderer
    .create(<Search text="" onChangeText={() => null} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders search with search string text', () => {
  const tree = renderer
    .create(<Search text="CRS-17" onChangeText={() => null} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
