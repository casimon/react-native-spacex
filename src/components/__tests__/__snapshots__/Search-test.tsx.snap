// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders search with empty search string 1`] = `
<View
  style={
    Array [
      Object {
        "padding": 16,
        "paddingTop": 22,
      },
      undefined,
    ]
  }
>
  <TextInput
    allowFontScaling={false}
    autoCompleteType="off"
    autoCorrect={false}
    onChangeText={[Function]}
    placeholder="Search launches e.g CRS-17"
    placeholderTextColor="#666"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#ccc",
        "borderRadius": 20,
        "height": 40,
        "paddingHorizontal": 30,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;

exports[`renders search with search string text 1`] = `
<View
  style={
    Array [
      Object {
        "padding": 16,
        "paddingTop": 22,
      },
      undefined,
    ]
  }
>
  <TextInput
    allowFontScaling={false}
    autoCompleteType="off"
    autoCorrect={false}
    onChangeText={[Function]}
    placeholder="Search launches e.g CRS-17"
    placeholderTextColor="#666"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#ccc",
        "borderRadius": 20,
        "height": 40,
        "paddingHorizontal": 30,
      }
    }
    underlineColorAndroid="transparent"
    value="CRS-17"
  />
</View>
`;
